hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -910,26 +914,25 @@ public void update(HoodieCleanMetadata cleanMetadata, String instantTime) {
   public void update(HoodieRestoreMetadata restoreMetadata, String instantTime) {
     dataMetaClient.reloadActiveTimeline();
 
-    // Since the restore has completed on the dataset, the latest write timeline instant is the one to which the
-    // restore was performed. If this is not present, then the restore was performed to the earliest commit. This
-    // could happen in case of bootstrap followed by rollback e.g. TestBootstrap#testFullBootstrapWithRegexModeWithUpdatesMOR.
-    Option<HoodieInstant> restoreInstant = dataMetaClient.getActiveTimeline().getWriteTimeline().lastInstant();
-    final String restoreToInstantTime;
-    if (restoreInstant.isPresent()) {
-      restoreToInstantTime = restoreInstant.get().getTimestamp();
-    } else {
-      restoreToInstantTime = metadataMetaClient.getActiveTimeline().filterCompletedInstants().firstInstant().get().getTimestamp();
+    // Fetch the commit to restore to (savepointed commit time)
+    HoodieInstant restoreInstant = new HoodieInstant(HoodieInstant.State.REQUESTED, HoodieTimeline.RESTORE_ACTION, instantTime);
+    HoodieInstant requested = HoodieTimeline.getRestoreRequestedInstant(restoreInstant);
+    HoodieRestorePlan restorePlan = null;
+    try {
+      restorePlan = TimelineMetadataUtils.deserializeAvroMetadata(
+          dataMetaClient.getActiveTimeline().readRestoreInfoAsBytes(requested).get(), HoodieRestorePlan.class);
+    } catch (IOException e) {
+      throw new HoodieIOException("Deserialization of rollback plan failed ", e);
```

Comment:
Good to have the restoreInstant in this log so can debug which plan is missing/corrupted.